ariadne/asgi.py

```diff
@@ -79,7 +80,11 @@ def __init__(
 
     async def get_context_for_request(self, request: Any) -> Any:
         if callable(self.context_value):
-            return self.context_value(request)
+            context = self.context_value(request)
+            if isawaitable(context):
```

Comment:
We could make it flat, guard-check for _Awaitable_ before this _Callable_ *if* statement. No real reason to keep them together.

Reply:
You can't do it like that - you need to call a function to get coroutine object that you can then test with `isawaitable` and then send to event loop via `await` to get its value.